### ISSUE TITLE
Control characters are being echo'ed back as user inputs 

### ISSUE THREAD

**Issue by jianyunt** [dotnet/corefx#29669]
Background
------------
In PowerShell Core (pwsh) on `Linux` or Azure Cloudshell (bash-Linux), when doing demos, we are consistently getting `OC`, `OD` characters magically inserted as a part of user inputs while pressing LeftArrow or RightArrow keys.  The situation occurs even more often when the PowerShell cmdlets display verbose messages or output messages, or running background jobs.

Sample
-------
 ``` C#
  //mimicking PowerShell executes a cmdlet
   Console.TreatControlCAsInput = true;
   var i = 100;
   while (i-- > 0)
   {
	Thread.Sleep (50);
	Console.WriteLine ("Working on it ...");
   }
  
   //PowerShell sets the cursor to begining after done with the execution
    var left = Console.CursorLeft;
    var top = Console.CursorTop;
    Console.WriteLine ("Left={0}, Top={1}", left, top);

    
    ConsoleKeyInfo input;
    do {
        input = Console.ReadKey(true);
         Console.WriteLine("{0}", input.Key);
    } while (input.Key.ToString() != "Q");


```
  
Sample Output
------
Press leftarrow key when you see "Working on it" message and then Q to quit. You will "O" and "D" instead of LeftArrow key.
![image](https://user-images.githubusercontent.com/10171679/39949895-b10bce30-5532-11e8-8190-e8ce84dffe4b.png)

Repro through PowerShell
-------------------
1. Linux VM
2. Install the PowerShell core from https://github.com/PowerShell/powershell#get-powershell

or https://portal.azure.com/

3. pwsh
4. We use the following script for doing something like background auth:
         $timer = New-Object Timers.Timer
         $timer.Interval=500
         $timer.Enabled=$true
         foreach ($a in 1..5){Register-ObjectEvent -InputObject $timer -EventName Elapsed -SourceIdentifier $a -Action {Start-Sleep -milliseconds 100}

5. Type any cmdlet e.g., Get-Module cmdlet  
6. Press Left or Right arrow
7. You may see
![image](https://user-images.githubusercontent.com/10171679/39949568-59828ebc-5530-11e8-9dfe-e8f85e6f73d4.png)

Initial Investigation
------------
When PowerShell completes a cmdlet execution, it sets the cursor to beginning by calling Console.CursorLeft and Console.CursorTop which in turn calls into  GetCursorPosition()  https://github.com/dotnet/corefx/blob/master/src/System.Console/src/System/ConsolePal.Unix.cs#L372.
Because GetCursorPostion() stripped out the ESC  (^[OD) for example, but put OD into the  _unprocessdBufferToBeRead buffer, so next time when Console.ReadKey() gets called, we are getting OD instead of the multiple character terminal sequence.

**Comment by karelz** [comment]
Does it happen also on .NET Core 2.1? (RC was on Monday)

**Comment by karelz** [comment]
It seems related to #25916 (dupe of #28694)

**Comment by jianyunt** [comment]
@karelz, yes it's happening in the latest release. Echo'ing control keys back in linux terminal  ^[OC, may not such big problem for us, for example, bash does the similar. However losing ESC and echoing back with consolekey code like OC, OD etc. as user inputs is the problem. Currently this issue is considered as blocking for `PowerShell Azure CloudShell` GA. Please take priority for the fix. Thanks.
CC @SteveL-MSFT, @edyoung  @HemantMahawar

**Comment by karelz** [comment]
@jianyunt can you please link your master tracking bug on your side? When is your GA? Is it a regression, or was it just discovered late?
cc @Petermarcu @leecow 

@joperezr @ianhays any ideas for workarounds / solutions here to unblock PowerShell?

**Comment by danmosemsft** [comment]
Marking 2.1 as potential candidate for early 2.1 servicing per shiproom discussion given impact on Powershell. @joperezr or @ianhays could one of you please investigate?

**Comment by danmosemsft** [comment]
@joperezr is going to confirm that Curses avoids this problem and if so try to figure out how. The goal being to see whether there is something we can do here that is small enough to do in servicing.

> Currently this issue is considered as blocking for PowerShell Azure CloudShell GA

@jianyunt can you confirm that Powershell 6.0 GA already shipped successfully with this bug present - is it somehow worse in Powershell Azure CloudShell - why does your team consider it ship blocking?

**Comment by jianyunt** [comment]
Yes. PowerShell 6.0 GA already shipped.  I think our Azure CloudShell is going to put PowerShell core on Linux in a real production environment. We are possibly the first customer of PS 6.0 on Linux.  Both Azure and PowerShell users are going to use it for sure.   I don't know other customer scenarios of PowerShell 6.0 on Linux. @joeyaiello?

**Comment by danmosemsft** [comment]
fixed

**Comment by karelz** [comment]
Fixed in 2.1 servicing in PR #29923. Part of 2.1.1 release.

**Comment by jianyunt** [comment]
👍
